host/sgx/create.c

```diff
@@ -832,6 +858,12 @@ oe_result_t oe_sgx_build_enclave(
         &loaded_enclave_pages_size,
         &enclave_size));
 
+    if (_is_kss_supported())
```

Comment:
This is my understanding from @bodzhang and @CodeMonkeyLeet's comments:

``c`
    if (_is_kss_supported())
    {
        // Set SECS.ATTRIBUTES.KSS bit if the CPU supports KSS
        context->attributes.flags |= OE_ENCLAVE_FLAG_KSS;
    }
    else if (props.config.attributes & OE_SGX_FLAGS_KSS)
	{
        // Fail if the CPU does not support KSS and the enclave specifies the KSS flag
        context->attributes.flags |= OE_ENCLAVE_FLAG_KSS;
		OE_RAISE_MSG(...);
    }
````

Reply:
Add it in oe_create_enclave, thanks

Reply:
Thought about the code a little more, flipping the KSS support bit on _is_kss_supported() is problematic. It's going to generate different enclave measurement. Here's the revised version:

````
    if (props.config.attributes & OE_SGX_FLAGS_KSS)
    {
        if ((context.type == OE_SGX_LOAD_TYPE_CREATE) && !_is_kss_supported())
        {
            // Fail if the CPU does not support KSS and the enclave specifies the KSS flag
            OE_RAISE_MSG(...);
        }

        context->attributes.flags |= OE_ENCLAVE_FLAG_SGX_KSS;
    }
````

Reply:
Cool. Perfect way to use context!  Updated

Reply:
SECS.ATTRIBUTES is not part of the measurement, so slipping the OE_ENCLAVE_FLAG_SGX_KSS bit in `context->attributes.flags` should not cause problem with enclave measurement. But the code update in @yentsanglee suggestion matches the design better and is required for a different reason - when performing enclave signing, KSS support on the underlining system should not be required. (my assumption is that the code flow is used for both enclave loading and enclave signing).